### PR TITLE
sql: Grant mz_introspection USAGE on schemas/dbs

### PR DESCRIPTION
**Pull request by jkosh44**
This commit adds new default privileges to give the mz_introsepection role USAGE privileges on all schemas and databases. It also grants USAGE privileges to mz_introspection on all existing databases and schemas.

These privileges make it easier to use debugging tools via the mz_introspection role and matches the default privileges on clusters.

Works towards resolving #20478

### Motivation
This PR adds a known-desirable feature.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

**Comment by mjibson** [comment]
I'm not convinced this is correct for the `mz_introspection` user, which was designed to not have access to user objects. We shouldn't change that just for 20478, unless there's wider organization buy in to letting the introspection user do these new things.

**Comment by jkosh44** [comment]
> I'm not convinced this is correct for the `mz_introspection` user, which was designed to not have access to user objects. We shouldn't change that just for 20478, unless there's wider organization buy in to letting the introspection user do these new things.

Ah I misunderstood and thought the concern was just with reading/explaining catalog items, not with using database/schema objects as well. It's not entirely new since `mz_introspection` can use all user clusters, which is needed for the Prometheus exporter. So if we don't want this change we should probably revert that as well. The two approaches I see are

1) Revoke all privileges on user clusters from `mz_introspection` and add some `mz_user_something` role that has `USAGE` privileges on all user clusters, schemas, and databases. The `mz_user_something` role will be used for the Prometheus exporter, Teleport, and any other tool that needs access to user objects.

2) Keep `mz_introspection` and give it `USAGE` privileges on schemas and databases. Continue using `mz_introspection` for Prometheus, Teleport, and whatever else it's used for.

I don't really have a strong opinion one way or the other, I heard @chaas and @benesch discussing this in the office but don't know if they reached a conclusion. I'll leave it to you three to decide.

For reference, here is the original feature request for `mz_introspection`: https://github.com/MaterializeInc/materialize/issues/15520

**Comment by benesch** [comment]
> I don't really have a strong opinion one way or the other, I heard @chaas and @benesch discussing this in the office but don't know if they reached a conclusion. I'll leave it to you three to decide.

Will respond on Slack.

File: src/adapter/src/catalog/storage/stash.rs [hunk 1]
```diff
@@ -228,6 +228,30 @@ pub async fn initialize(
                 privileges: AclMode::USAGE,
             },
         ),
+        (
+            DefaultPrivilegesKey {
+                role_id: RoleId::Public,
```
Comment (mjibson) [inline src/adapter/src/catalog/storage/stash.rs:233]: `role_id` is the role doing the granting, not the role gain the privilege, right?

**Review by mjibson — APPROVED** [review]
This still doesn't allow EXPLAIN because that also requires SELECT?

**Comment by jkosh44** [comment]
> This still doesn't allow EXPLAIN because that also requires SELECT?

That's correct.

File: src/adapter/src/catalog/storage/stash.rs (hunk 1 above)
Comment (jkosh44) [reply src/adapter/src/catalog/storage/stash.rs:233]: That's correct. That's the name PostgreSQL uses in their docs, though I'd be open to changing it to something more obvious.

**Pull request merged**